### ISSUE TITLE
Uncontrolled Checkbox not responding to keyboard as it should

### ISSUE THREAD

**Issue by chriscorwin** [salesforce/design-system-react#753]
Checkboxen ought to be able to be focused by `tab` (works) and then toggled via spacebar (not working in uncontrolled version, **works in controlled, of course, if the parent component provides it, as it does in the indeterminate story**).

I've also asked the SLDS team about checkbox markup, because the [docs have conflicting advice](https://salesforceux.slack.com/archives/slds-ux/p1478014973000747).
